src/hooks/usePoolData.ts

```diff
@@ -334,3 +355,32 @@ export default function usePoolData(
 
   return poolData
 }
+
+async function getSgtApy(
+  rewardsContract: SharedStakeStakingRewards | null,
+  sgtPrice = 0,
+  tvlUsd: BigNumber,
+): Promise<BigNumber> {
+  // https://github.com/SharedStake/SharedStake-ui/blob/main/src/components/Earn/geyser.vue#L336
+  if (rewardsContract == null || tvlUsd.eq(Zero)) return Zero
+  const now = BigNumber.from(Math.floor(Date.now() / 1000))
+  const [until, rewardsDuration, sgtRewardsPerPeriod] = await Promise.all([
+    rewardsContract.periodFinish(), // 1e0 timestamp in seconds
+    rewardsContract.rewardsDuration(), // 1e0 seconds
+    rewardsContract.getRewardForDuration(), // 1e18
```

Comment:
nit: this would be a great place to use multicall if possible so it's only 1 request